formats/json-jackson/src/main/java/io/cloudevents/jackson/PojoCloudEventDataMapper.java

```diff
@@ -37,12 +40,17 @@ private PojoCloudEventDataMapper(ObjectMapper mapper, TypeReference<T> target) {
         try {
             value = this.mapper.readValue(bytes, this.target);
         } catch (Exception e) {
-            throw CloudEventRWException.newDataConversion(e, target.getType().toString());
+            throw CloudEventRWException.newDataConversion(e, target.getTypeName());
         }
         return new PojoCloudEventData<>(mapper, value, bytes);
     }
 
     public static <T> PojoCloudEventDataMapper<T> from(ObjectMapper mapper, TypeReference<T> target) {
```

Comment:
Can you add javadoc to this method and the other one, explaining a bit the difference between class and typereference?